### PR TITLE
Make PaC Webhook work on KCP

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### CODE REVIEW

File: controllers/component_build_controller.go [hunk 1]
```diff
@@ -173,9 +175,20 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		log.Info("Pipelines as Code enabled")
 
 		// Obtain Pipelines as Code callback URL
-		webhookTargetUrl, err := r.getPaCRoutePublicUrl(ctx)
-		if err != nil {
-			return ctrl.Result{}, err
+		webhookTargetUrl := os.Getenv(pipelinesAsCodeRouteEnvVar)
+		if webhookTargetUrl == "" {
+			// The env variable is not set
+			if req.ClusterName != "" {
```
Comment (gabemontero) [inline controllers/component_build_controller.go:181]:
```suggestion
			if req.ClusterName == "" {
```

if the string is empty, then it is not set, right?

**Review by gabemontero — CHANGES_REQUESTED** [review]

File: controllers/component_build_controller.go (hunk 1 above)
Comment (mmorhun) [reply controllers/component_build_controller.go:181]: The comment is related to the if above: `if webhookTargetUrl == "" ` that happens after an attempt to read the env variable.

Comment (mmorhun) [reply controllers/component_build_controller.go:181]: I'll add an empty line to separate

**Review by Michkov — APPROVED** [review]

File: controllers/component_build_controller.go (hunk 1 above)
Comment (gabemontero) [reply controllers/component_build_controller.go:182]: ah ok I see making this resolved

**Review by gabemontero — APPROVED** [review]
/lgtm

**Pull request merged**